server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java

```diff
@@ -531,13 +534,13 @@ public RestoreSnapshotRequest source(Map<String, Object> source) {
             } else if (name.equals("rename_pattern")) {
                 if (entry.getValue() instanceof String) {
                     renamePattern((String) entry.getValue());
-                } else {
+                } else if (entry.getValue() != null) {
```

Comment:
if im reading this right, it will only fail if its not a string and is not null. Otherwise itll emit null below in the `toXContent`. I think the null checks would make sense in `toXContent` as well, but we certainly dont want a non-string being emitted, so i think the IAE is ok here.

Reply:
@hub-cap you're reading it right. However, I decided to revert to the old behavior and to check in `#toXContent()` to skip emitting `renamePattern` / `renameReplacement` when they are `null`.